Changed saveServerIcon() to not use request() on defaultIconUrl

Swapnilr1 opened the pull request:
---
<!--
Remove the fields that are not appropriate
Please include:
-->

**What's this PR do?**
Fixes https://github.com/zulip/zulip-electron/issues/533

**Any background context you want to provide?**

**Screenshots?**

**You have tested this PR on:**
  - [ ] Windows
  - [x] Linux/Ubuntu
  - [ ] macOS

abhigyank reviewed (approved):
@akashnimare This looks good to me and works as expected, could be merged.
Though the issue itself is not much of a problem to us now cause we fallback to the initial character of the realm, but this would now correctly prevent the error from propping up.

Swapnilr1 commented:
I made a mistake while squashing the commits together, so this is 2 commits instead of 1. I hope this is fine. @akashnimare Could you review this?

Swapnilr1 committed:
Changed saveServerIcon() to not use request() on defaultIconUrl

context-menu: Fix context menu indexing.

Addresses a problem where the context menu would get the incorrect index
of server in sidebar and cause a crash when attempting to remove the
server.

Add comment justifying extra index parameter.

Swapnilr1 committed:
Merge remote-tracking branch 'upstream/master'

Swapnilr1 commented:
Please discard this. I will open up a new PR.

Swapnilr1 committed:
Revert "Changed saveServerIcon() to not use request() on defaultIconUrl"

This reverts commit dd1ce9cb01e04b72ae43627dde04c51708a4e141.
